code/game/gamemodes/blob/blobs/shield.dm

```diff
@@ -19,3 +19,32 @@
 /obj/structure/blob/shield/CanPass(atom/movable/mover, turf/target, height=0)
 	if(istype(mover) && mover.checkpass(PASSBLOB))	return 1
 	return 0
+
+/obj/structure/blob/shield/reflective
+	name = "reflective blob"
+	desc = "A solid wall of slightly twitching tendrils with a reflective glow."
+	icon_state = "blob_idle_glow"
+	brute_resist = 0
+	health = 50
+	maxHealth = 50
+	flags_1 = CHECK_RICOCHET_1
+	var/reflect_chance = 80 //80% chance to reflect
+
+/obj/structure/blob/shield/reflective/handle_ricochet(obj/item/projectile/P)
+	if(P.is_reflectable && prob(reflect_chance) && !istype(P, /obj/item/projectile/beam/pulse))
+		var/P_turf = get_turf(P)
+		var/face_direction = get_dir(src, P_turf)
+		var/face_angle = dir2angle(face_direction)
+		var/incidence_s = GET_ANGLE_OF_INCIDENCE(face_angle, (P.Angle + 180))
+		if(abs(incidence_s) > 90 && abs(incidence_s) < 270)
+			return FALSE
+		var/new_angle_s = SIMPLIFY_DEGREES(face_angle + incidence_s)
+		P.Angle = new_angle_s

```

Comment:
Make a setangle proc just like tg, skipping the part in their setangle proc which we don't use